Add Registration form

pavan-testpress opened the pull request:
### Changes done
- Added Registration form.
- Added Email verfication.
- Added Signup View.
- Written TestCases.

### Reason for the changes
- To Register new user and verify their email address.

### Stats
- Number of Test Cases - 4 
- Number of SQL Queries - 4

### Guidelines
- [x] PEP8(Python)
- [x] Is each logical block separated with a line break?
- [x] Have you self reviewed this PR in context to the previous PR feedbacks?

pavan-testpress committed:
Add Registration Form

pavan-testpress committed:
Add Registration Form

- Add Registration templates
- Use UserCreationForm
- Add signup view
- Add Email verfication Feature
- Add TestCases

bharathwaaj commented on onboarding/apps/accounts/views.py:
```diff
@@ -1,7 +1,59 @@
+    if request.user.is_authenticated:
+        return HttpResponseRedirect(reverse('accounts:home'))
+    else:
+        if request.method == 'POST':
```
A better way to write would be
```
form = UserCreationForm(data = request.POST or None)
if form.is_valid():
    # Write code for POST operation
    return HttpResponseRedirect(success_page)

# Write code for GET operation
return render(request...)
```

bharathwaaj commented on onboarding/apps/accounts/views.py:
```diff
@@ -1,7 +1,59 @@
+                mail_subject = 'Activate your account.'
+                to_email = form.cleaned_data.get('email')
+                send_mail(mail_subject, message, 'pavan1995143.pavan@gmail.com', [to_email, ])
+                return HttpResponse('Please confirm your email address to complete the registration')
```
A POST operation should always be followed by a redirect.

bharathwaaj replied:
https://en.m.wikipedia.org/wiki/Post/Redirect/Get

bharathwaaj commented on onboarding/tests/test_accounts/test_signup.py:
```diff
@@ -0,0 +1,60 @@
+        Test to create user through signup page.
+        """
+        response = self.client.post(reverse('accounts:signup'), self.user)
+        self.assertEqual(response.context['domain'], 'testserver')
```
This test might change if sign up redirects after POST.

bharathwaaj commented on onboarding/tests/test_accounts/test_signup.py:
```diff
@@ -0,0 +1,60 @@
+        self.assertEqual(response.context['form'].errors['username'], ['Please verify your email to login.'])
+
+    def test_to_activte_user(self):
+        """
```
activate spelling

bharathwaaj commented on onboarding/tests/test_accounts/test_signup.py:
```diff
@@ -0,0 +1,60 @@
+        """
+        response = self.client.post(reverse('accounts:signup'), self.user)
+        self.assertEqual(response.context['domain'], 'testserver')
+        self.assertEqual(response.context['uid'], urlsafe_base64_encode(force_bytes(response.context['user'].pk)).decode())
```
How is uid available in context ? It defeats the purpose of uid verification

bharathwaaj commented on onboarding/tests/test_accounts/test_signup.py:
```diff
@@ -0,0 +1,60 @@
+        self.assertEqual(response.status_code, 200)
+        user = User.objects.get(pk=response.context['user'].pk)
+        self.assertEqual(user.is_active, False)
+        activate = self.client.get(reverse('accounts:account_activate', kwargs={'uidb64': response.context['uid'], 'token': response.context['token']}))
```
Same here

pavan-testpress commented on onboarding/apps/accounts/views.py:
```diff
@@ -1,7 +1,59 @@
+        user.save()
+        my_login(request, user)
+        return HttpResponse('<span>Thank you for your email confirmation. Now you can <a href="/login/">login</a> your account.</span>')
+    else:
```
If user tries to open the same link again it will display that else part message. don't we need that ?

pavan-testpress committed:
Do Code Enhancements

- Use alias names for inbuilt forms.
- Redirect Post request.
- typo corrections.
- if else code enchancements.
